### PR TITLE
[SYCL-MLIR][ArgumentPromotion] Fix incorrect behaviour when first argument is not a candidate

### PR DESCRIPTION
**Pull request by whitneywhtsang**
The variable `pos` is used to determine the position of the arguments.
It is only incremented by new arguments size, but it should also be incremented by original argument.

### CODE REVIEW

File: polygeist/test/polygeist-opt/sycl/argpromotion_invalid.mlir
Comment (whitneywhtsang) [inline polygeist/test/polygeist-opt/sycl/argpromotion_invalid.mlir:1]: no need to review this file, part of https://github.com/intel/llvm/pull/8897.

File: polygeist/test/polygeist-opt/sycl/argpromotion.mlir
Comment (whitneywhtsang) [inline polygeist/test/polygeist-opt/sycl/argpromotion.mlir:1]: only added callee5 and test5 on top of https://github.com/intel/llvm/pull/8897.

**Review by whitneywhtsang — COMMENTED** [review]
This PR depends on https://github.com/intel/llvm/pull/8897.

**Review by etiotto — APPROVED** [review]
LGTM - thanks!

**Pull request merged**